### PR TITLE
Add back `Get-Counter` cmdlet for Windows

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Reviewed the APIs being used.  `Get-Counter` can work with just public APIs published in pdh.h.  `Import-Counter` and `Export-Counter` rely on undocumented APIs so those are removed.  The PDH apis are really old so I don't believe there's any chance to get those APIs or the binary log format documented so removed that code.  `Get-Counter -ListSet` has Description property as empty because it isn't supported by PDH apis.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5020
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Diagnostics/CommonUtils.cs [hunk 1]
```diff
@@ -14,6 +14,130 @@ namespace Microsoft.PowerShell.Commands.Diagnostics.Common
 {
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Diagnostics/CommonUtils.cs]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Get-Counter; ConfirmImpact=; SideIndicator==>}, @{Name=Tee-Object; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

**Comment by sdwheeler** [comment]
@SteveL-MSFT https://go.microsoft.com/fwlink/?linkid=2109647

File: src/Microsoft.PowerShell.Commands.Diagnostics/CommonUtils.cs [hunk 2]
```diff
@@ -14,9 +14,90 @@ namespace Microsoft.PowerShell.Commands.Diagnostics.Common
...
     internal static class CommonUtilities
     {
+        private const string LibraryLoadDllName = "api-ms-win-core-libraryloader-l1-2-0.dll";
+        private const string LocalizationDllName = "api-ms-win-core-localization-l1-2-1.dll";
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Diagnostics/CommonUtils.cs]: Do this work on Windows 7/8?

File: src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs [hunk 3]
```diff
@@ -31,7 +31,7 @@ namespace Microsoft.PowerShell.Commands
     ///
     /// Class that implements the Get-Counter cmdlet.
     ///
-    [Cmdlet(VerbsCommon.Get, "Counter", DefaultParameterSetName = "GetCounterSet", HelpUri = "https://go.microsoft.com/fwlink/?LinkID=138335")]
+    [Cmdlet(VerbsCommon.Get, "Counter", DefaultParameterSetName = "GetCounterSet", HelpUri = "https://go.microsoft.com/fwlink/?LinkID=2109647")]
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs]: Should we update a test for the new FWLink code?

**Review by iSazonov — COMMENTED** [review]
I found https://github.com/microsoft/Tx. It seems work with blg.

File: src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs (hunk 3 above)
Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs]: Will add

**Comment by SteveL-MSFT** [comment]
@iSazonov nice find, seems that Tx is using a different set of APIs to read the log, not sure if there is a different public api to write the binary log.  I suggest we defer this for now.

**Review by iSazonov — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@iSazonov created https://github.com/PowerShell/PowerShell/issues/10946

**Comment by TravisEz13** [comment]
@SteveL-MSFT Can you run a coordinated package build and send me a Direct Message with the build information?  Let's verify this doesn't cause any new compliance issues.

File: src/Microsoft.PowerShell.Commands.Diagnostics/Microsoft.PowerShell.Commands.Diagnostics.csproj [hunk 4]
```diff
@@ -15,19 +15,19 @@
...
+
+  <ItemGroup Condition=" '$(IsWindows)' != 'true' ">
+    <Compile Remove="GetCounterCommand.cs" />
     <Compile Remove="CounterSample.cs" />
     <Compile Remove="CounterSet.cs" />
-    <Compile Remove="ExportCounterCommand.cs" />
-    <Compile Remove="GetCounterCommand.cs" />
-    <Compile Remove="GetEventSnapin.cs" />
-    <Compile Remove="ImportCounterCommand.cs" />
     <Compile Remove="PdhHelper.cs" />
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Diagnostics/Microsoft.PowerShell.Commands.Diagnostics.csproj]: Compliance issue.  Communicated offline

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Diagnostics/CommonUtils.cs [hunk 5]
```diff
@@ -14,9 +14,90 @@ namespace Microsoft.PowerShell.Commands.Diagnostics.Common
 {
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Diagnostics/CommonUtils.cs]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-macos`
ConvertTo-Json.Should handle 'AutomationNull.Value' and 'NullString.Value' correctly
```powershell
Expected strings to be the same, but they were different.
Expected length: 37
Actual length:   33
Strings differ at index 14.
Expected: '{"a":null,"b":null,"c":null,"d":null}'
But was:  '{"a":null,"b":{},"c":null,"d":{}}'
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1: line 79
79:         } | ConvertTo-Json -Compress | Should -BeExactly '{"a":null,"b":null,"c":null,"d":null}'
```
ConvertTo-Json.Should handle the ETS properties added to 'DBNull.Value' and 'NullString.Value'
```powershell
Expected strings to be the same, but they were different.
Expected length: 69
Actual length:   56
Strings differ at index 36.
Expected: '..."},{"value...'
But was:  '..."},{"nulls...'
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1: line 98
98:             $p1, $p2 | ConvertTo-Json -Compress | Should -BeExactly '[{"value":null,"dbnull":"dbnull"},{"value":null,"nullstr":"nullstr"}]'
```

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**